### PR TITLE
[#1124] docs(tez): Add the document for tez-client

### PR DESCRIPTION
**Pull request by bin41215**
### What changes were proposed in this pull request?

Add the document for tez-client.

### Why are the changes needed?

Fix: #1124 

### Does this PR introduce _any_ user-facing change?

No.

### CODE REVIEW

File: docs/client_guide.md [hunk 1]
```diff
@@ -242,6 +242,7 @@ Notice: this feature requires the MEMORY_LOCAL_HADOOP mode.
 |--------------------------------|---------|-------------------------------------------------------------------------|
 | tez.rss.avoid.recompute.succeeded.task | false   | Whether to avoid recompute succeeded task when node is unhealthy or black-listed |
 | tez.rss.client.max.buffer.size | 3k | The max buffer size in map side |
+| tez.rss.client.batch.trigger.num | 50 | The max batch of buffers to send data in map side |
```
Comment (jerqi) [inline docs/client_guide.md:245]: Could you give more detailed description? What will it happen if the task reach the value?

File: docs/client_guide.md [hunk 2]
```diff
@@ -242,6 +242,7 @@ Notice: this feature requires the MEMORY_LOCAL_HADOOP mode.
 |--------------------------------|---------|-------------------------------------------------------------------------|
 | tez.rss.avoid.recompute.succeeded.task | false   | Whether to avoid recompute succeeded task when node is unhealthy or black-listed |
 | tez.rss.client.max.buffer.size | 3k | The max buffer size in map side |
```
Comment (jerqi) [inline docs/client_guide.md:244]: Could you modify the decription, too?

**Comment by bin41215** [comment]
ok

**Review by jerqi — APPROVED** [review]
LGTM. thanks @bin41215

**Pull request merged**